src/strand.rs

```diff
@@ -0,0 +1,306 @@
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match *self {
+            Strand::Unknown => Ok(()),
+            _ => write!(f, "({})", self.strand_symbol()),
+        }
+    }
+}
+
+impl From<ReqStrand> for Strand {
+    fn from(rstr: ReqStrand) -> Self {
+        match rstr {
+            ReqStrand::Forward => Strand::Forward,
+            ReqStrand::Reverse => Strand::Reverse,
+        }
+    }
+}
+
+/// Strand information for annotations that require a strand.
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Ord, PartialOrd, Copy)]
+pub enum ReqStrand {
+    Forward,
+    Reverse,
+}
+
+impl Same for ReqStrand {
+    fn same(&self, s1: &Self) -> bool {
+        self == s1
+    }
+}
+
+impl Display for ReqStrand {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "({})", self.strand_symbol())
+    }
+}
+
+impl Strandedness for ReqStrand {
+    fn reverse(&self) -> Self {
+        match *self {
+            ReqStrand::Forward => ReqStrand::Reverse,
+            ReqStrand::Reverse => ReqStrand::Forward,
+        }
+    }
+
+    fn try_req_strand(&self) -> Option<ReqStrand> {
+        Some(*self)
+    }
+
+    fn strand_symbol(&self) -> &str {
+        match *self {
+            ReqStrand::Forward => "+",
+            ReqStrand::Reverse => "-",
+        }
+    }
+
+    fn break_pos_strand(posstr: &str) -> Option<(&str, Self)> {
+        if posstr.ends_with("(+)") {
+            if let Some(pos) = posstr.get(0..(posstr.len() - 3)) {
+                return Some((pos, ReqStrand::Forward));
+            }
+        } else if posstr.ends_with("(-)") {
+            if let Some(pos) = posstr.get(0..(posstr.len() - 3)) {
+                return Some((pos, ReqStrand::Reverse));
+            }
+        }
+
+        None
+    }
+}
+
+impl KnownStrandedness for ReqStrand {
+    fn req_strand(&self) -> ReqStrand {
+        *self
+    }
+}
+
+/// Strand information for annotations that definitively have no
+/// strand information.
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Ord, PartialOrd, Copy)]
+pub enum NoStrand {
+    Unknown,
+}
+
+impl Same for NoStrand {
+    fn same(&self, _s1: &Self) -> bool {
+        true
+    }
+}
+
+impl Display for NoStrand {
+    fn fmt(&self, _f: &mut Formatter) -> fmt::Result {
+        Ok(())
+    }
+}
+
+impl Strandedness for NoStrand {
+    fn reverse(&self) -> Self {
+        NoStrand::Unknown
+    }
+    fn try_req_strand(&self) -> Option<ReqStrand> {
+        None
+    }
+    fn strand_symbol(&self) -> &str {
+        "."
+    }
+    fn break_pos_strand(posstr: &str) -> Option<(&str, Self)> {
+        Some((posstr, NoStrand::Unknown))
+    }
+}
+
+/// Equality-like operator for comparing strand information. Unknown
+/// strands are not equal, but they are the "same" as other unknown
+/// strands.
+pub trait Same {
+    /// Indicate when two strands are the "same" -- two
+    /// unknown/unspecified strands are the "same" but are not equal.
+    fn same(&self, &Self) -> bool;
```

Comment:
I think this is the perfect use case for just implementing [PartialEq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html).

Reply:
I found situations where I wanted to test true equality (i.e., keep Unknown on an equal footing). Even in the test code, this does come up.